### PR TITLE
Moving reconcile/testing from serving to pkg.

### PR DESCRIPTION
**Pull request by n3wscott**
Related to https://github.com/knative/eventing/issues/734

### CODE REVIEW

File: reconciler/testing/clock.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2018 The Knative Authors.
```
Comment (mattmoor-sockpuppet) [inline reconciler/testing/clock.go]:
```suggestion
Copyright 2019 The Knative Authors.
```

File: reconciler/testing/configmap.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/ghodss/yaml"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// TODO(mattmoor): Move this into `knative/pkg/configmap`
```
Comment (mattmoor) [inline reconciler/testing/configmap.go]: Wanna do this? :D

File: reconciler/testing/reactions.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	}
+}
+
+func ValidateCreates(action clientgotesting.Action) (handled bool, ret runtime.Object, err error) {
+	got := action.(clientgotesting.CreateAction).GetObject()
+	obj, ok := got.(apis.Validatable)
+	if !ok {
+		return false, nil, nil
+	}
+	if err := obj.Validate(context.Background()); err != nil {
```
Comment (mattmoor) [inline reconciler/testing/reactions.go]: Can you open an issue to get the right context decorators plumbed into these?  

This one is each: `apis.WithinCreate(context.Backgroundm())`, but below is hard `apis.WithinUpdate(context.Background(), ???)`

File: reconciler/testing/configmap.go (hunk 2 above)
Comment (n3wscott) [reply reconciler/testing/configmap.go]: as a followup

Comment (n3wscott) [reply reconciler/testing/configmap.go]: done.

**Pull request merged**